### PR TITLE
Issue 30815 populate the unique fields table when the database validation is enabled

### LINKED ISSUE #30815: Populate the unique_fields table when the Database validation is enabled
### Parent Issue

_No response_

### User Story

Currently, a TaskUpgrade is used to populate the table. However, if validation is disabled and later re-enabled, the TaskUpgrade will need to be executed again. To address this, we should ensure the table is repopulated every time validation is enabled. This requires detecting changes to the validation property and triggering the population logic accordingly.

Additionally, we need to enhance the existing logic (currently in the TaskUpgrade) with the following updates:

- Include the Variant.
- Add a boolean field to indicate whether the version is live. (Currently, only one version is being included.)
- Include a boolean field to store the value of the uniquePerSite field variable.

### Acceptance Criteria

Populate the unique_fields table every time the Data Base Unique Field Validation is enabled
Remove the Task241007CreateUniqueFieldsTable class

### Steps to Reproduce

* In your `docker-compose.yml` file, set the folowing env var:
```yml
DOT_FEATURE_FLAG_DB_UNIQUE_FIELD_VALIDATION: true
```
* Start up dotCMS with the Full Starter.
  * Once it's up and running, connect to the database instance. In your terminal, you can run `docker ps` to get the name of the database Docker image, and then connect to it like this: `docker exec -it {{DOCKER-IMAGE-NAME}} /bin/bash`
* Get the total record count on the `unique_fields` table. You must get the following result:
```
dotcms=# select count(*) from unique_fields;
 count 
-------
   236
(1 row)
```
* Now, bring dotCMS down and disable the flag:
```yml
DOT_FEATURE_FLAG_DB_UNIQUE_FIELD_VALIDATION: false
```
* Start dotCMS again and, when it's up and running, connect to the database instance and get the total count on the `unique_fields` table. You must get the following result:
```
dotcms=# select count(*) from unique_fields;
ERROR:  relation "unique_fields" does not exist
LINE 1: select count(*) from unique_fields;
```
* Re-enable the flag, start dotCMS, and make sure that the `unique_fields` table is back again.

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by freddyDOTCMS**
### Proposed Changes
* Create a Initializer to check if the Database validation was enabled 

https://github.com/dotCMS/core/pull/30862/files#diff-b1a4fb7917e1a3e9e98ded9a8d7f31ffd2b90a7aa873c73ba2a8f35b82282a12R17

* Create a Method to create the table 
https://github.com/dotCMS/core/pull/30862/files#diff-b0aed2eddcc36be2a2e2f46624345a83c8698db238fe4dc228c0fb13a11e344fR419

* Create a method to drop the table

https://github.com/dotCMS/core/pull/30862/files#diff-b0aed2eddcc36be2a2e2f46624345a83c8698db238fe4dc228c0fb13a11e344fR440

* Create a method to populate the table

https://github.com/dotCMS/core/pull/30862/files#diff-b0aed2eddcc36be2a2e2f46624345a83c8698db238fe4dc228c0fb13a11e344fR464

* Add WrapInTransaction and  CloseDBIfOpened annotation to each method in UniqueFieldDataBaseUtil

https://github.com/dotCMS/core/pull/30862/files#diff-b0aed2eddcc36be2a2e2f46624345a83c8698db238fe4dc228c0fb13a11e344fR138-R353

This PR fixes: #30815

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/UniqueFieldsValidationInitializer.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.util.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+import java.sql.SQLException;
+
+/**
+ * Initializer in charge of check when dotMCS start up if the Unique Fields Data Base validation was enabled
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/UniqueFieldsValidationInitializer.java:17]: dotCMS*

Comment (freddyDOTCMS) [reply dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/UniqueFieldsValidationInitializer.java]: done dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/UniqueFieldsValidationInitializer.java

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/UniqueFieldsValidationInitializer.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * It check if the table already exists and:
+ * - If it exists and the Database validation is disabled then drop the table.
+ * - If it does not exist and the Database validation is enabled then it created and populate it.
+ * - If it exists and the Database validation is enabled do nothing.
+ * - If it does not exist  and the Database validation is disabled do nothing.
+ */
+@Dependent
+public class UniqueFieldsValidationInitializer  implements DotInitializer {
+
+    private UniqueFieldDataBaseUtil uniqueFieldDataBaseUtil;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/UniqueFieldsValidationInitializer.java:29]: set to final

File: dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/extratable/UniqueFieldDataBaseUtil.java [hunk 3]
```diff
@@ -220,6 +274,7 @@ private static String getUniqueRecalculationQuery(final boolean uniquePerSite) {
                 uniquePerSite);
     }
 
+    @CloseDBIfOpened
     public List<Map<String, Object>> get(final String contentId, final long languegeId) throws DotDataException {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/uniquefields/extratable/UniqueFieldDataBaseUtil.java:278]: typo here, may be languageId

**Review by jdotcms — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**